src/hash_table/mod.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Amari Robinson
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+pub mod chaining;
+
+use core::cmp;
+
+/// Hash Table Load Factor
+#[derive(Clone, Copy, Debug)]
+pub enum LoadFactor {
+    /// A chaining hash table.
+    Chained(ChainedLoadFactor),
+    /// An "open addressing" hash table.
+    OpenAddressed(OpenAddressedLoadFactor),
```

Comment:
This doesn't seem to ever be used? Why not simplify it and only have `ChainedLoadFactor`?

Reply:
Are array-based/openly-addressed hash tables are outside the scope of `intrusive-rs`? 

I have a [SwissTable](https://abseil.io/about/design/swisstables) (sans the SIMD) implementation on a private branch.

I have a few use cases where a fixed-size openly addressed hash table is better suited than a chaining implementation:
- variants of event / message dispatch where
  - zero-alloc context/handler state
  - the context/handler state is a value embedded into the hash table:
    - intrusive collection: list, heap, rbtree
    - keys for another associative map

Reply:
I believe that would be outside the scope. The only reason you would need any kind of intrusive link would be if you had a linked list in a chained hash table. It is not needed for an open-addressed hash table.